database/sql/V057__add_fee_policy.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TYPE PolicyKind AS ENUM ('priceimprovement', 'volume');
+
+CREATE TABLE fee_policies (
+  auction_id bigint NOT NULL,
+  order_uid bytea NOT NULL,
+  -- The order in which the fee policies are inserted and applied.
+  application_order SERIAL NOT NULL,
```

Comment:
Is SERIAL correct here? Shouldn't this just be the index of the fee policy for the given (auction, order) pair?

Reply:
I've put SERIAL so that it would autoincrement every time we insert the same `<auction_id, order_uid>`. How would you version with a simple index look like?

Reply:
auction_id | order_uid | application_order
-- | -- | --
1 | 1 | 1
1 | 1 | 2
2 | 2 | 3

should probably become

auction_id | order_uid | application_order
-- | -- | --
1 | 1 | 1
1 | 1 | 2
2 | 2 | 1

That would require manual indexing within the (auction, order). Correct me if I am wrong.

Reply:
Manual indexing adds more complexity to the code. @fleupold, let me know if I got the intention correctly, so I'll be able to continue with that or just skip it.

Reply:
I'm fine with both (although the second one is prettier). What is important is that the ordering is right. Meaning, given a pair <auction_id, order_id> give me all the policies ordered the same way as they were applied (application_order)

Reply:
I think the first option also gives us an opportunity to get ordered policies just by a given order (not even using auction_ids). Not sure if we going to need this option though.

Reply:
I see (I thought option 2 was the intended one, but it's true that 1 works as well). No strong feelings, whatever is simpler to implement.